### PR TITLE
[Merged by Bors] - feat: add some iteratedDeriv variants of iteratedDerivWithin lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Based on work from EulerMaclaurin.

Co-authored-by: Geoffrey Irving <irving@naml.us>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean [hunk 1]
```diff
@@ -108,14 +141,12 @@ theorem iteratedDeriv_const_smul {n : ℕ} {f : 𝕜 → F} (h : ContDiff 𝕜 n
       ← Function.comp_def, deriv.scomp x h₀ (differentiableAt_id'.const_mul _),
       deriv_const_mul _ differentiableAt_id', deriv_id'', smul_smul, mul_one, pow_succ]
 
+/-- Note: this is unrelated to `iteratedDerivWithin_const_mul`, where the multiplication works on
+the codomain. -/
 theorem iteratedDeriv_const_mul {n : ℕ} {f : 𝕜 → 𝕜} (h : ContDiff 𝕜 n f) (c : 𝕜) :
```
Comment (eric-wieser) [inline Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean:146]: Hmm, maybe this should be called `iteratedDeriv_comp_const_mul`?

File: Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean [hunk 2]
```diff
@@ -91,6 +95,35 @@ theorem iteratedDerivWithin_sub (hf : ContDiffOn 𝕜 n f s) (hg : ContDiffOn 
...
+    iteratedDerivWithin_add (Set.mem_univ _) uniqueDiffOn_univ
+      (contDiffOn_univ.mpr hf) (contDiffOn_univ.mpr hg)
+
+theorem iteratedDeriv_const_add (hn : 0 < n) (c : F) :
+    iteratedDeriv n (fun z => c + f z) x = iteratedDeriv n f x := by
+  simpa only [iteratedDerivWithin_univ] using
+    iteratedDerivWithin_const_add (Set.mem_univ _) uniqueDiffOn_univ hn c
+
+theorem iteratedDeriv_const_neg (hn : 0 < n) (c : F) :
+    iteratedDeriv n (fun z => c - f z) x = iteratedDeriv n (-f) x := by
```
Comment (eric-wieser) [inline Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean:110]: This is a `sub` theorem not a `neg` theorem

File: Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean:146]: I'll do this in a followup

File: Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean (hunk 2 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean]: So it is. Fixed

**Pull request closed without merging**